synapse/rest/admin/rooms.py

```diff
@@ -292,6 +292,39 @@ async def on_GET(
         return 200, ret
 
 
+class RoomStateRestServlet(RestServlet):
+    """
+    Get full state within a room.
+    """
+
+    PATTERNS = admin_patterns("/rooms/(?P<room_id>[^/]+)/state")
+
+    def __init__(self, hs: "HomeServer"):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+        self.message_handler = hs.get_message_handler()
+
+    async def on_GET(
+        self, request: SynapseRequest, room_id: str
+    ) -> Tuple[int, JsonDict]:
+        requester = await self.auth.get_user_by_req(request)
+        await assert_user_is_admin(self.auth, requester.user)
+
+        ret = await self.store.get_room(room_id)
+        if not ret:
+            raise NotFoundError("Room not found")
+
+        room_state = await self.message_handler.get_state_events(
+            user_id=requester.user.to_string(),
+            room_id=room_id,
+            is_admin=True,  # already verified above
+        )
```

Comment:
It's probably easier (and more efficient) to do:

```python
event_ids = await self.store.current_state_events(room_id)
events = await self.store.get_events(event_ids)
now = self.clock.time_msec()
events = await self._event_serializer.serialize_events(
    room_state.values(),
    now,
    # We don't bother bundling aggregations in when asked for state
    # events, as clients won't use them.
    bundle_aggregations=False,
)
```

Reply:
> event_ids = await self.store.current_state_events(room_id)

This function doesn't seem to exist?

Reply:
Gah, bad c+p, sorry. It's meant to be `self.store.get_current_state_ids`